pkg/cmd/server/api/types.go

```diff
@@ -968,6 +968,11 @@ type KubernetesMasterConfig struct {
 	// the server will not start. These values may override other settings in KubernetesMasterConfig which may cause invalid
 	// configurations.
 	ControllerArguments ExtendedArguments
+
+	// ReconcileMasterEndpoints indicates that OpenShift should update the list of endpoints for the
+	// "kubernetes" Service, removing any entry from the list whose corresponding master/apiserver is
+	// unhealthy or down.
+	ReconcileMasterEndpoints bool
```

Comment:
@smarterclayton has previously stated a desire, at least for Kube, if this had gone upstream-first, for this to be disabled by default so that we could get some bake-in time before we enable it all the time. If we want to enable this by default, I'm happy to make that change (although how do we handle the fact that we're adding a new bool that needs to default to true in the config?)

I do plan on filing issues/PRs for openshift-ansible and openshift-docs for this feature.